lib/spack/spack/compilers/__init__.py

```diff
@@ -156,7 +156,15 @@ def get_compiler_config_from_packages(
 def _compiler_config_from_package_config(config):
     compilers = []
     for entry in config:
-        compiler = _compiler_config_from_external(entry)
+        try:
+            compiler = _compiler_config_from_external(entry)
+        except Exception as e:
+            msg = "Reading compiler from packages config section failed\n"
```

Comment:
this should report the file and line from the specific file.  we have that info in the config.